arch/xtensa/core/cpu_idle.c

```diff
@@ -6,48 +6,14 @@
 #include <zephyr/toolchain.h>
 #include <zephyr/tracing/tracing.h>
 
-/* xt-clang removes any NOPs more than 8. So we need to set
- * no optimization to avoid those NOPs from being removed.
- *
- * This function is simply enough and full of hand written
- * assembly that optimization is not really meaningful
- * anyway. So we can skip optimization unconditionally.
- * Re-evalulate its use and add #ifdef if this assumption
- * is no longer valid.
- */
+#ifndef CONFIG_ARCH_CPU_IDLE_CUSTOM
 __no_optimization
```

Comment:
Remove `__no_optimization` as it is no longer needed.

Reply:
sure, thanks!